File: adapters/java/I18nt.java
package i18n;

import java.text.MessageFormat;
import java.util.Map;
import java.util.HashMap;

/**
 * i18nt Java Helper Class
 */
public class I18nt {
    private final Map<String, Object> data;

    public I18nt(Map<String, Object> data) {
        this.data = data;
    }

    /**
     * 获取翻译文本
     * @param path 点分路径，如 "auth.login.title"
     */
    public String t(String path) {
        return t(path, null);
    }

    /**
     * 获取翻译文本并进行插值
     * @param path 点分路径
     * @param params 插值参数
     */
    @SuppressWarnings("unchecked")
    public String t(String path, Map<String, Object> params) {
        String[] parts = path.split("\\.");
        Object current = data;
        
        for (String part : parts) {
            if (current instanceof Map) {
                current = ((Map<String, Object>) current).get(part);
            } else {
                return "[" + path + "]";
            }
        }

        if (current instanceof String) {
            String template = (String) current;
            if (params != null) {
                // 将 {name} 转换为 MessageFormat 兼容的 {0} 格式
                int i = 0;
                Object[] values = new Object[params.size()];
                for (Map.Entry<String, Object> entry : params.entrySet()) {
                    template = template.replace("{" + entry.getKey() + "}", "{" + i + "}");
                    values[i++] = entry.getValue();
                }
                return MessageFormat.format(template, values);
            }
            return template;
        }
        
        return current != null ? current.toString() : "[" + path + "]";
    }
}

File: adapters/kotlin/I18nt.kt
package i18n

import java.text.MessageFormat

/**
 * i18nt Kotlin Helper Class
 */
class I18nt(private val data: Any?) {

    /**
     * 支持点分路径访问：t("auth.login")
     */
    fun t(path: String, params: Map<String, Any>? = null): String {
        val parts = path.split(".")
        var current: Any? = data
        
        for (part in parts) {
            current = (current as? Map<*, *>)?.get(part)
            if (current == null) return "[$path]"
        }

        return format(current, params) ?: "[$path]"
    }

    /**
     * 支持方括号访问：t["auth"]["login"]
     */
    operator fun get(key: String): I18nt {
        val subData = (data as? Map<*, *>)?.get(key)
        return I18nt(subData)
    }

    /**
     * 直接调用对象获取字符串：t["key"].toString()
     */
    override fun toString(): String {
        return (data as? String) ?: ""
    }

    private fun format(template: Any?, params: Map<String, Any>?): String? {
        if (template !is String) return template?.toString()
        if (params == null) return template

        var result = template
        val values = mutableListOf<Any>()
        params.entries.forEachIndexed { index, entry ->
            result = result.replace("{${entry.key}}", "{$index}")
            values.add(entry.value)
        }
        
        return try {
            MessageFormat.format(result, *values.toTypedArray())
        } catch (e: Exception) {
            result
        }
    }
}

/**
 * 扩展函数：让 Map 直接支持 I18nt 转换
 */
fun Map<String, Any>.asI18n() = I18nt(this)
